aion/test/auth/auth_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule Aion.AuthTest do
+  use Aion.ConnCase
+  alias Aion.User
+
+  @user %User{name: "name", email: "email", password: "password", id: 5}
+
+  test "login should set current_user, add user_id to session and set plug_session_info to :renew", %{conn: conn} do
+    conn = conn
+     |> Plug.Test.init_test_session(foo: "bar")
+     |> Aion.Auth.login(@user)
+
+    assert conn.assigns.current_user == @user
+    assert conn.private.plug_session["user_id"] == @user.id
+    assert conn.private.plug_session_info == :renew
+  end
+
+  test "login_by_username_and_pass should return :ok when passwords match", %{conn: conn} do
+    conn = Plug.Test.init_test_session(conn, foo: "bar")
+    user = User.registration_changeset(%User{}, Map.from_struct(@user))
+    Aion.Repo.insert(user)
+
+    login_result = Aion.Auth.login_by_username_and_pass(conn, @user.email, @user.password, repo: Aion.Repo)
+
+    assert tuple_size(login_result) == 2
```

Comment:
Just for the future, I think in general we could match the exact results (if it's possible), cause just checking for list's/tuple's length may not be the most accurate thing. 

This is what we've been doing so far, so no need to fix it, just suggesting that we could take another approach in the future.

Reply:
I wanted to do it but it was not possible since authenticated conn contains things that depend on time (I think some kind of timestamp).

Second thing is that the result is a tuple that is `{:ok, login(conn, user)}` and `login` method was tested above. Also its the only result that provides tuple that has length 2 so I think its fine (although not that scalable).